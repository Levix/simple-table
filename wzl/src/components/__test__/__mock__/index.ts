import type { TableColumns, TableDataSourceType } from '../../types';
import { useDataSource } from "../../../composables/useDataSource";
import { useColumns } from "../../../composables/useColumns";
import { ref } from 'vue';

const dataSource: TableDataSourceType[] = []
for (let index = 0; index < 10; index++) {
    dataSource.push({
        key: index,
        name: `Edrward ${index}`,
        age: 18 + index,
        address: `London Park no. ${index}`,
        tags: ['nice', 'developer'],
    })
}
// 表格源数据
export { dataSource };
// 表格列配置
export const columns: TableColumns[] = [
    {
		title: "Name",
		key: "name",
	},
	{
		title: "Age",
		key: "age",
		width: 100,
		align:'center',
		sortable: {
			orderBy: "DESC",
			sorter: (curr, next) => next - curr,
		},
	},
	{
		title: "Address",
		key: "address",
	},
	{
		title: "Tags",
		key: "tags",
	},
];

export const sortIconObj = {
    ascTriangle: "▲",
    descTriangle: "▼",
    normalTriangle: "■",
}

export const a_columns: TableColumns[] = [
    {
        title: "Col_1",
        key: "Col__1",
    },
    {
        title: "Col_2",
        key: "Col__2",
        sortable: {
            orderBy: "asc"
        }
    },
    {
        title: "Col_3",
        key: "Col__3",
    },
    {
        title: "Col_4",
        key: "Col__4",
    },
]
export const d_columns: TableColumns[] = [
    {
        title: "Col_1",
        key: "Col__1",
    },
    {
        title: "Col_2",
        key: "Col__2",
        sortable: {
            orderBy: "desc"
        }
    },
    {
        title: "Col_3",
        key: "Col__3",
    },
    {
        title: "Col_4",
        key: "Col__4",
    },
]

// 处理后的配置
export const { newColumns } = useColumns(columns);
// 处理后的数据
export const { sortTableData } = useDataSource(ref(dataSource), newColumns);
// body配置
export const bodyOptions = {
    sortData: sortTableData.value,
    columns: newColumns.value,
    sortOption: {
        key: 1,
        sortStatus: 'asc',
    },
    pageSize: 10,
    curPageNumber: 1,
}

// foot配置
export const footOptions = {
    sortData: sortTableData.value,
    pageSize: 4,
}

export const newFootOptions = {
    sortData: sortTableData.value,
    pageSize: 8,
}