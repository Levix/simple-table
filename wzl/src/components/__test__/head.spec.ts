import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import WuHead from '../Head';
import { newColumns, a_columns, d_columns, sortIconObj } from './__mock__/index';

describe('WuHead', () => {
    const wrapper = mount(WuHead, {
        props: {
            columns: newColumns.value,
        }
    })

    it('打个快照 WuHead', () => {
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.exists()).toBe(true);
    })

    it('组件是否正常渲染，参数是否正确', async () => {
        expect(wrapper.find('thead').exists()).toBe(true);

        expect(wrapper.findAll('th')).toHaveLength(newColumns.value.length);

        // 改变columns
        await wrapper.setProps({
            columns: a_columns
        })
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find('span').exists()).toBe(true);

        await wrapper.setProps({
            columns: d_columns
        })
        
        // 点击事件是否触发
        const iconWrapper = wrapper.find('span');

        await iconWrapper.trigger('click');
        expect(wrapper.text()).toContain(sortIconObj.normalTriangle);

        await iconWrapper.trigger('click');
        expect(wrapper.text()).toContain(sortIconObj.ascTriangle);

        await iconWrapper.trigger('click');
        expect(wrapper.text()).toContain(sortIconObj.descTriangle);
    })
})