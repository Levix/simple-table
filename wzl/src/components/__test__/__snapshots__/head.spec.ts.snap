// Vitest Snapshot v1

exports[`WuHead > 打个快照 WuHead 1`] = `
"<thead class=\\"wu_head\\">
  <tr>
    <th title=\\"Name\\" class=\\"align__left width__200px\\">Name
      <!---->
    </th>
    <th title=\\"Age\\" class=\\"align__center width__100px\\">Age<span class=\\"wu_head__sort\\">▼</span></th>
    <th title=\\"Address\\" class=\\"align__left width__200px\\">Address
      <!---->
    </th>
    <th title=\\"Tags\\" class=\\"align__left width__200px\\">Tags
      <!---->
    </th>
  </tr>
</thead>"
`;

exports[`WuHead > 组件是否正常渲染，参数是否正确 1`] = `
"<thead class=\\"wu_head\\">
  <tr>
    <th title=\\"Col_1\\" class=\\"\\">Col_1
      <!---->
    </th>
    <th title=\\"Col_2\\" class=\\"\\">Col_2<span class=\\"wu_head__sort\\">▲</span></th>
    <th title=\\"Col_3\\" class=\\"\\">Col_3
      <!---->
    </th>
    <th title=\\"Col_4\\" class=\\"\\">Col_4
      <!---->
    </th>
  </tr>
</thead>"
`;
